### ISSUE TITLE
Feature Request: Upgrade to use the kubernetes 1.22 client libraries

### ISSUE THREAD

**Issue by spencer-p** [kubernetes-sigs/cli-utils#415]
Attempting to use cli-utils@v0.25.0 (the latest release at time of writing) with k8s.io/api@v0.22.2 results in the following:
```
 $ go mod tidy
go: finding module for package sigs.k8s.io/cli-utils/pkg/manifestreader
go: found sigs.k8s.io/cli-utils/pkg/manifestreader in sigs.k8s.io/cli-utils v0.25.0
go: finding module for package k8s.io/api/batch/v2alpha1
gke-internal.googlesource.com/addon-manager-v2/pkg/prune tested by
        gke-internal.googlesource.com/addon-manager-v2/pkg/prune.test imports
        sigs.k8s.io/cli-utils/pkg/manifestreader imports
        k8s.io/kubectl/pkg/cmd/util imports
        k8s.io/kubectl/pkg/scheme imports
        k8s.io/api/batch/v2alpha1: module k8s.io/api@latest found (v0.22.2), but does not contain package k8s.io/api/batch/v2alpha1
```

The master branch has a go.mod configured to use 1.21 libraries, but the latest release is still configured for 1.20 libraries. (Is the Github Actions bot broken?)

Running `go get sigs.k8s.io/cli-utils@master` fixes the build.

Expected behavior: Building a project with cli-utils@latest and k8s.io/api@latest succeeds.

**Comment by ash2k** [comment]
We should probably merge https://github.com/kubernetes-sigs/cli-utils/pull/390 at some point and cut a new version.

**Comment by karlkfi** [comment]
Updated in https://github.com/kubernetes-sigs/cli-utils/pull/435

This is in master and will be available when we release v0.27.0, planned for December 2021.
